src/screens/App/Stats/index.js

```diff
@@ -0,0 +1,338 @@
+		grid-template-columns: 1fr;
+	}
+`;
+
+const Card = styled('div')`
+	background-color: ${mediumGrey};
+	border-radius: 4px;
+	padding: 1rem;
+	min-height: 100px;
+	position: relative;
+`;
+
+const Number = styled(P)`
+	font-size: 3rem;
+	font-weight: 500;
+	color: ${accentGrey};
+	margin: 0;
+`;
+
+const TimeSelectContainer = styled('div')`
+	display: flex;
+	align-items: center;
+	width: 200px;
+`;
+
+const TimeSelect = styled(ArianneElem)`
+	margin-left: 5px;
+	width: 100%;
+`;
+
+const MetaHeading = styled('div')`
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+
+	@media (max-width: ${BREAKPOINTS}px) {
+		flex-direction: column;
+		align-items: start;
+	}
+`;
+
+const PageSubHeading = styled(SubHeading)`
+	margin: 2rem 0;
+`;
+
+const Section = styled('div')`
+	margin-bottom: 5rem;
+`;
+
+const Stats = ({history, location}) => {
+	const {
+		data: {
+			me: {tasks},
+		},
+		error,
+	} = useQuery(GET_ALL_TASKS, {suspend: true});
+	const {
+		data: {
+			me: {startWorkAt, endWorkAt},
+		},
+	} = useQuery(GET_USER_INFOS, {suspend: true});
+
+	const diffTime = moment(endWorkAt, 'HH:mm:ss').diff(
+		moment(startWorkAt, 'HH:mm:ss'),
+		'hours',
+		true,
+	);
+	const workingTime = diffTime < 0 ? diffTime + 24 : diffTime;
+
+	const query = new URLSearchParams(location.search);
+
+	const since = parseInt(query.get('since'), 10) || 30;
+	const projectId = query.get('projectId');
+	const tags = query.getAll('tags');
+	const linkedCustomerId = query.get('customerId');
+
+	if (error) throw error;
+
+	const setSince = useCallback(
+		(value) => {
+			const newQuery = new URLSearchParams(query);
+
+			newQuery.set('since', value);
+
+			history.push(`/app/stats?${newQuery.toString()}`);
+		},
+		[query],
+	);
+
+	const setProjectSelected = (selected, removeCustomer) => {
+		const newQuery = new URLSearchParams(query);
+
+		if (selected) {
+			const {value: selectedProjectId} = selected;
+
+			newQuery.set('projectId', selectedProjectId);
+		}
+		else if (newQuery.has('projectId')) {
+			newQuery.delete('projectId');
+		}
+
+		if (removeCustomer) {
+			newQuery.delete('customerId');
+		}
+
+		history.push(`/app/stats?${newQuery.toString()}`);
+	};
+
+	const setCustomerSelected = (selected) => {
+		const newQuery = new URLSearchParams(query);
+
+		if (selected) {
+			const {value: selectedCustomerId} = selected;
+
+			newQuery.set('customerId', selectedCustomerId);
+		}
+		else if (newQuery.has('customerId')) {
+			newQuery.delete('customerId');
+		}
+
+		if (newQuery.has('projectId')) {
+			newQuery.delete('projectId');
+		}
+
+		history.push(`/app/stats?${newQuery.toString()}`);
+	};
+
+	const setTagSelected = (selected) => {
+		const newQuery = new URLSearchParams(query);
+
+		if (selected) {
+			newQuery.delete('tags');
+			selected.forEach(tag => newQuery.append('tags', tag.value));
+		}
+
+		history.push(`/app/stats?${newQuery.toString()}`);
+	};
+
+	const filteredTasks = tasks.filter(
```

Comment:
I know that this does not change that much but maybe we should use a combination of useState and useEffect to make sure that the reference is not changed for unnecessary reason.

Reply:
don't know what the best practices are honestly but I think being careful around minimizing reference update could be good for the app in general tell me if I'm missing something.